cartography/intel/aws/s3.py

```diff
@@ -257,26 +257,52 @@ def _load_s3_acls(neo4j_session: neo4j.Session, acls: Dict, aws_account_id: str,
 
 
 @timeit
-def _load_s3_policies(neo4j_session: neo4j.Session, policies: List[Dict], update_tag: int) -> None:
+def _load_s3_policies(neo4j_session: neo4j.Session, policies: List[Dict], update_tag: int, boto3_session: boto3.session.Session) -> None:
     """
     Ingest S3 policy results into neo4j.
     """
+
+    bucketPolicies = add_bucket_policy_documents(policies, boto3_session)
+
     # NOTE we use the coalesce function so appending works when the value is null initially
     ingest_policies = """
     UNWIND $policies AS policy
     MATCH (s:S3Bucket) where s.name = policy.bucket
     SET s.anonymous_access = (coalesce(s.anonymous_access, false) OR policy.internet_accessible),
     s.anonymous_actions = coalesce(s.anonymous_actions, []) + policy.accessible_actions,
-    s.lastupdated = $UpdateTag
+    s.lastupdated = $UpdateTag,
+    s.policy_document = policy.policyDocument
     """
 
     neo4j_session.run(
         ingest_policies,
-        policies=policies,
-        UpdateTag=update_tag,
+        policies=bucketPolicies,
+        UpdateTag=update_tag
     )
 
 
+@timeit
+def add_bucket_policy_documents(policies: List[Dict], boto3_session: boto3.session.Session) -> List[Dict]:
+    """
+    Fetch policy documents for each S3 bucket in the policies list and prepare a list with the necessary details.
+    """
+    bucket_policies = []
+    for policy in policies:
+        bucket_name = policy.get('bucket')
+        client = boto3_session.client('s3')
+        response = get_policy({'Name': bucket_name}, client)
+        policy_document = response.get('Policy', {}) if response else {}
+
+        bucket_policies.append({
```

Comment:
instead of creating new object can you add `policy_document` key direct to policy object

Reply:
Changes done!